packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -0,0 +1,439 @@
+    const [{ cartId }] = useCartContext();
+
+    const { data: countriesData } = useQuery(getAllCountriesQuery);
+    const { data: billingAddressData } = useQuery(getBillingAddressQuery, {
+        variables: { cartId }
+    });
+    const { data: shippingAddressData } = useQuery(getShippingAddressQuery, {
+        variables: { cartId }
+    });
+    const { data: isBillingAddressSameData } = useQuery(
+        getIsBillingAddressSameQuery,
+        { variables: { cartId } }
+    );
+    const [
+        updateBillingAddress,
+        {
+            error: billingAddressMutationErrors,
+            called: billingAddressMutationCalled,
+            loading: billingAddressMutationLoading
+        }
+    ] = useMutation(setBillingAddressMutation);
+    const [
+        updateCCDetails,
+        {
+            error: ccMutationErrors,
+            called: ccMutationCalled,
+            loading: ccMutationLoading
+        }
+    ] = useMutation(setCreditCardDetailsOnCartMutation);
+
+    const { countries } = countriesData || {};
+    const isBillingAddressSame = formState.values.isBillingAddressSame;
+
+    const initialValues = useMemo(() => {
+        const isBillingAddressSame = isBillingAddressSameData
+            ? isBillingAddressSameData.cart.isBillingAddressSame
+            : true;
+
+        let billingAddress = {};
+        if (billingAddressData) {
+            if (billingAddressData.cart.billingAddress) {
+                const {
+                    // eslint-disable-next-line no-unused-vars
+                    __typename,
+                    ...rawBillingAddress
+                } = billingAddressData.cart.billingAddress;
+                billingAddress = mapAddressData(rawBillingAddress);
+            }
+        }
+
+        return { isBillingAddressSame, ...billingAddress };
+    }, [isBillingAddressSameData, billingAddressData]);
+
+    const errors = useMemo(() => {
+        const errors = [];
+
+        if (ccMutationErrors) {
+            ccMutationErrors.graphQLErrors.forEach(({ message }) => {
+                errors.push(message);
+            });
+        }
+        if (billingAddressMutationErrors) {
+            billingAddressMutationErrors.graphQLErrors.forEach(
+                ({ message }) => {
+                    errors.push(message);
+                }
+            );
+        }
+
+        return errors;
+    }, [ccMutationErrors, billingAddressMutationErrors]);
+
+    /**
+     * Helpers
+     */
+
+    const setIsBillingAddressSameInCache = useCallback(() => {
+        client.writeQuery({
+            query: getIsBillingAddressSameQuery,
+            data: {
+                cart: {
+                    __typename: 'Cart',
+                    id: cartId,
+                    isBillingAddressSame
+                }
+            }
+        });
+    }, [client, cartId, getIsBillingAddressSameQuery, isBillingAddressSame]);
+
+    const setShippingAddressAsBillingAddress = useCallback(() => {
```

Comment:
Mentioned in the GraphQL file, seems we can greatly clean this up by syncing the two before you get to this step. Sorry for missing that in my scope 😐